### DISCUSSION TITLE
Rerun failed tests cause more tests executed.

### DISCUSSION THREAD

**Discussion by windyliu1118** [getgauge/gauge#2141] [General]
I executed the whole auto test suites in jennkins, and when I refer to the execution output, I found there are only 2 tests and 1 spec failed, but 4 tests and 3 specs are executed when using the `gauge run --failed`.
This is really weird, and for these months audit, this situation will always happen. 
My gauge version:
```
Gauge version: 1.4.1
Commit Hash: a82478a

Plugins
-------
html-report (4.0.10)
js (2.3.17)
screenshot (0.0.1)
xml-report (0.2.3)
```
taiko version is ```Version: 1.0.18 (Chromium: 86.0.4214.0) RELEASE```

And my shell script is as below:
```
#!/bin/bash


source ~/.nvm/nvm.sh

nvm use 10.16.3

export NODE_OPTIONS="--max-old-space-size=8192"


echo "remove previous reports and screenshots"

rm .gauge/screenshots/*
rm -r reports/*

npm install

gauge -v

taiko -v

echo "run auto tests"

gauge run --max-retries-count=2 --env staging_jenkins specs

if [ $? -eq 0 ]; then
  echo "All test pass at the running"
  exit 0
else
  echo "Rerun the failed tests"
  gauge run --failed
fi
```
1st executed result
<img width="988" alt="Screen Shot 2021-10-28 at 2 56 34 PM" src="https://user-images.githubusercontent.com/7007401/139203267-96e6994e-2cfb-4a65-8b7b-e82cce9d9e4b.png">

result of rerun the failed tests:
<img width="968" alt="Screen Shot 2021-10-28 at 2 56 43 PM" src="https://user-images.githubusercontent.com/7007401/139203325-0077e306-a7b8-4e52-b394-52828564d7d5.png">

Could you pls take a look and confirm why this issue happens? Thanks a lot.

**Comment by windyliu1118** [comment]
Can anyone take a look at this issue?

**Reply by sriv** [reply]
what does `failure.json` contain post the run?

if you have a sample project to replicate this issue, I can take a deeper look

**Comment by zabil** [comment]
Looks like this is related to https://github.com/getgauge/gauge/issues/2158

**Reply by windyliu1118** [reply]
yeah, maybe that's the issue. @zabil is there any plan to fix this?

**Reply by zabil** [reply]
There are no plans for fix this at the moment.

**Comment by windyliu1118** [comment]
@zabil for such issue https://github.com/getgauge/gauge/issues/2158, do we have any plan to fix it?

**Reply by zabil** [reply]
No plans to fix this at the moment. The issue is  labelled as "help wanted" in case anyone wants to pick it up for contribution.
